tribe-ext-settings-import-export.php

```diff
@@ -0,0 +1,370 @@
+			// $this->add_required_plugin( 'Tribe__Events__Filterbar__View', '4.3.3' );
+			// $this->add_required_plugin( 'Tribe__Events__Tickets__Eventbrite__Main', '4.3.2' );
+			// $this->add_required_plugin( 'Tribe_APM', '4.4' );
+
+			// Conditionally-require Events Calendar PRO. If it is active, run an extra bit of code.
+			//add_action( 'tribe_plugins_loaded', [ $this, 'detect_tec_pro' ], 0 );
+		}
+
+		/**
+		 * Check required plugins after all Tribe plugins have loaded.
+		 *
+		 * Useful for conditionally-requiring a Tribe plugin, whether to add extra functionality
+		 * or require a certain version but only if it is active.
+		 */
+		public function detect_tec_pro() {
+			if ( Tribe__Dependency::instance()->is_plugin_active( 'Tribe__Events__Pro__Main' ) ) {
+				$this->add_required_plugin( 'Tribe__Events__Pro__Main', '4.3.3' );
+				$this->ecp_active = true;
+			}
+		}
+
+		/**
+		 * Extension initialization and hooks.
+		 */
+		public function init() {
+			// Load plugin textdomain
+			// Don't forget to generate the 'languages/tribe-ext-settings-import-export.pot' file
+			load_plugin_textdomain( 'PLUGIN_TEXT_DOMAIN', false, basename( dirname( __FILE__ ) ) . '/languages/' );
+
+			if ( ! $this->php_version_check() ) {
+				return;
+			}
+
+			// Insert filters and hooks here
+			//add_filter( 'thing_we_are_filtering', [ $this, 'my_custom_function' ] );
+			add_action( 'admin_menu', array( $this, 'tribe_settings_menu' ), 99 );
+            add_action( 'admin_init', array( $this, 'tribe_sie_process_settings_export' ) );
+			add_action( 'admin_init', array( $this, 'tribe_sie_process_settings_import' ) );
+			add_action( 'admin_init', array( $this, 'tribe_sie_process_settings_reset' ) );
+		}
+
+		/**
+		 * Check if we have a sufficient version of PHP. Admin notice if we don't and user should see it.
+		 *
+		 * @link https://theeventscalendar.com/knowledgebase/php-version-requirement-changes/ All extensions require PHP 5.6+.
+		 *
+		 * @return bool
+		 */
+		private function php_version_check() {
+			$php_required_version = '5.6';
+
+			if ( version_compare( PHP_VERSION, $php_required_version, '<' ) ) {
+				if (
+					is_admin()
+					&& current_user_can( 'activate_plugins' )
+				) {
+					$message = '<p>';
+					$message .= sprintf( __( '%s requires PHP version %s or newer to work. Please contact your website host and inquire about updating PHP.', 'PLUGIN_TEXT_DOMAIN' ), $this->get_name(), $php_required_version );
+					$message .= sprintf( ' <a href="%1$s">%1$s</a>', 'https://wordpress.org/about/requirements/' );
+					$message .= '</p>';
+					tribe_notice( 'PLUGIN_TEXT_DOMAIN' . '-php-version', $message, [ 'type' => 'error' ] );
+				}
+				return false;
+			}
+			return true;
+		}
+
+		/**
+		 * Register the settings page
+		 */
+		function tribe_settings_menu() {
+			//add_options_page( __( 'Sample Settings Import and Export' ), __( 'Sample Settings Import and Export' ), 'manage_options', 'tribe-settings', 'tribe_settings_page' );
```

Comment:
Remove for tidiness! 

Where possible, use version control to store old code rather than leaving it in place (but commented out).